src/assets/wise5/classroomMonitor/dataExport/strategies/AbstractComponentDataExportStrategy.ts

```diff
@@ -0,0 +1,309 @@
+   * @param componentState the component state
+   * @return an array containing the cells in the row
+   */
+  createStudentWorkExportRow(
+    rowCounter: number,
+    workgroupId: number,
+    userId1: number,
+    userId2: number,
+    userId3: number,
+    studentName1: string,
+    studentName2: string,
+    studentName3: string,
+    periodName: string,
+    componentRevisionCounter: ComponentRevisionCounter,
+    componentState: any
+  ): string[] {
+    const row = new Array(this.columnNames.length);
+    row.fill('');
+    row[this.columnNameToNumber.get('#')] = rowCounter;
+    row[this.columnNameToNumber.get('Workgroup ID')] = workgroupId;
+    this.setStudentIDsAndNames(
+      row,
+      userId1,
+      studentName1,
+      userId2,
+      studentName2,
+      userId3,
+      studentName3
+    );
+    row[this.columnNameToNumber.get('Class Period')] = periodName;
+    row[this.columnNameToNumber.get('Project ID')] = this.configService.getProjectId();
+    row[this.columnNameToNumber.get('Project Name')] = this.projectService.getProjectTitle();
+    row[this.columnNameToNumber.get('Run ID')] = this.configService.getRunId();
+    row[this.columnNameToNumber.get('Student Work ID')] = componentState.id;
+    row[
+      this.columnNameToNumber.get('Server Timestamp')
+    ] = this.utilService.convertMillisecondsToFormattedDateTime(componentState.serverSaveTime);
+    const clientSaveTime = new Date(componentState.clientSaveTime);
+    const clientSaveTimeString =
+      clientSaveTime.toDateString() + ' ' + clientSaveTime.toLocaleTimeString();
+    row[this.columnNameToNumber.get('Client Timestamp')] = clientSaveTimeString;
+    row[this.columnNameToNumber.get('Node ID')] = componentState.nodeId;
+    row[this.columnNameToNumber.get('Component ID')] = componentState.componentId;
+    row[this.columnNameToNumber.get('Step Title')] = this.projectService.getNodePositionAndTitle(
+      componentState.nodeId
+    );
+    const componentPartNumber =
+      this.projectService.getComponentPosition(componentState.nodeId, componentState.componentId) +
+      1;
+    row[this.columnNameToNumber.get('Component Part Number')] = componentPartNumber;
+    row[this.columnNameToNumber.get('Component Type')] = this.component.content.type;
+    if (this.component.content.prompt != null) {
+      let prompt = this.utilService.removeHTMLTags(this.component.content.prompt);
+      prompt = prompt.replace(/"/g, '""');
+      row[this.columnNameToNumber.get('Component Prompt')] = prompt;
+    }
+    const studentData = componentState.studentData;
+    if (studentData != null) {
+      row[this.columnNameToNumber.get('Student Data')] = studentData;
+      const isCorrect = studentData.isCorrect;
+      if (isCorrect != null) {
+        if (isCorrect) {
+          row[this.columnNameToNumber.get('Is Correct')] = 1;
+        } else {
+          row[this.columnNameToNumber.get('Is Correct')] = 0;
+        }
```

Comment:
Replace with ternary for brevity (5 lines vs 1)?
```suggestion
        row[this.columnNameToNumber.get('Is Correct')] = isCorrect ? 1 : 0;
```